src/movies.js

```diff
@@ -1,22 +1,78 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+  const directors = moviesArray.map(element => element.director)
+  return directors;
+};
+
+console.log(getAllDirectors(movies));
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+        
+  const dramaSpielberg = moviesArray.filter((element) =>  element.director === "Steven Spielberg" &&
+  element.genre.includes("Drama"));
+
+  
+    return dramaSpielberg.length;
+                                        
+};
+
+console.log(howManyMovies(movies));
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+if (moviesArray.length === undefined){
+  return 0;
+} else {
+  let score = moviesArray.reduce( function(acumulator, currentValue){
```

Comment:
you need to check if score exist for every movies or not :
inside reduce method you can write:
 if(!currentValue.score){
        currentValue.score = 0;
        return acumulator  + currentValue.score ;
      }else{
        return acumulator  + currentValue.score ;
      }